README.md

```diff
@@ -124,6 +124,46 @@ implemented as a union in Avro, where `null` is the first type in the union.
 `fixed` and `enum` fields may be specified inline as part of a record
 or as standalone named types.
 
+```ruby
+# Either syntax is supported for specifying the size
+fixed :f, 4
+fixed :g, size: 8
+
+# Either syntax is supported for specifying symbols
+enum :e, :X, :Y, :Z
+enum :d, symbols: [:A, :B]
+
+record :my_record_with_named do
+  required :f_ref, :f
+  required :fixed_inline, :fixed, size: 9
+  required :e_ref, :e
+  required :enum_inline, :enum, symbols: [:P, :Q]
+end
+```
+
+### Unions
+
+A union may be specified within a record:
+
+```ruby
+record :my_record do
+  union :my_union, :string, :int
```

Comment:
When declared like this are these required or optional? Should we drop this syntax and only use the `required` / `optional` syntax mentioned later on to be consistent with other fields?